### PR TITLE
Program details refactoring

### PR DESCRIPTION
**Pull request by nkrivous**

### CODE REVIEW

File: packages/shared/components/programs/program-details/services/program-details.service.ts [hunk 1]
```diff
@@ -62,7 +71,9 @@ export const getProgramStatistic = (
   });
 };
 
-export const closeProgram = (programId, opts) => dispatch => {
+export const closeProgram = (programId: string, opts: any) => (
```
Comment (pavelzubov) [inline packages/shared/components/programs/program-details/services/program-details.service.ts]: Add opts type

File: packages/shared/components/programs/program-details/services/program-details.service.ts [hunk 2]
```diff
@@ -90,7 +103,11 @@ export const closePeriod = (programId, onSuccess) => dispatch => {
     });
 };
 
-export const fetchProgramTrades = (id, filters, currency) => {
+export const fetchProgramTrades = (
+  id: string,
+  filters: any,
```
Comment (pavelzubov) [inline packages/shared/components/programs/program-details/services/program-details.service.ts]: Add filters type

File: packages/shared/components/programs/program-details/services/program-details.service.ts [hunk 3]
```diff
@@ -104,7 +121,7 @@ export const fetchProgramTrades = (id, filters, currency) => {
     });
 };
 
-export const fetchOpenPositions = (id, filters) => {
+export const fetchOpenPositions = (id: string, filters: any) => {
```
Comment (pavelzubov) [inline packages/shared/components/programs/program-details/services/program-details.service.ts]: Add filters type

Comment (pavelzubov) [reply packages/shared/components/programs/program-details/services/program-details.service.ts]: and output type

File: packages/shared/components/programs/program-details/services/program-details.service.ts (hunk 2 above)
Comment (pavelzubov) [reply packages/shared/components/programs/program-details/services/program-details.service.ts]: and output type

File: packages/investor/src/modules/program-reinvesting/components/program-reinvesting-container.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  IProgramReinvestingContainerState
+> {
+  constructor(props: IProgramReinvestingContainerProps) {
+    super(props);
+    this.state = {
+      isReinvesting: props.isReinvesting,
+      isPending: false
+    };
+  }
+  toggleReinvesting = (updateDetails: any) => (value: boolean) => {
```
[the remark this answers is not in the document]
Comment (nkrivous) [reply packages/investor/src/modules/program-reinvesting/components/program-reinvesting-container.tsx]: done

**Pull request merged**